common/transport/amqp/src/amqp.ts

```diff
@@ -254,12 +254,14 @@ export class Amqp {
               done(err);
             };
 
+            this._amqp.on('client:errorReceived', attachErrorHandler);
```

Comment:
>this._amqp.on('client:errorReceived', attachErrorHandler); [](start = 12, length = 58)

Are you actually gaining anything here.  Are you going to have to face handling an error twice?  Will the error first come through on the receiver, and then be "resignalled" via the client?

Reply:
Yup, you don't get these errors in the links because they aren't attached yet. You want the links removed of the dictionary because you're going to hit 'disconnecting' next which is going to try and forceDetach the links, which will screw things up.
There is probably something to be done to simplify error management but amqp10 doesn't make it easy on us for now this passes all the tests :) 

---
In reply to: [140017498](https://github.com/Azure/azure-iot-sdk-node/pull/85#discussion_r140017498) [](ancestors = 140017498)